uqcsbot/scripts/attic.py

```diff
@@ -0,0 +1,128 @@
+from uqcsbot import bot, Command
+from uqcsbot.util.status_reacts import loading_status
+from typing import List
+import os
+import re
+import requests
+
+
+BASE_FOLDER_URL = 'https://drive.google.com/drive/folders/'
+BASE_FILE_URL = 'https://drive.google.com/file/d/'
+BASE_API_URL = 'https://www.googleapis.com/drive/v3/'
+BASE_ATTIC_FOLDER = '0B6_D4T6LJ-uwZmFhMzIyNGYtNTM2OS00ZDJlLTg0NmYtY2IyNzA1MDZlNDIx'
+API_KEY = os.environ['GOOGLE_API_KEY']
+ROOM_FILE_LIMIT = 15  # Number of files allowed to be posted in room, else sent via direct message.
+SLACK_MAX_LINES = 45  # Slack appears to split messages past this point.
+COURSE_REGEX = r'[A-Z]{4}[1-7]{1}[0-9]{3}'
+
+
+def format_files(files: List[dict], course: dict) -> List[str]:
+    """
+    Takes the list of file dictionaries found and formats them into Slack-appropriate message strings.
+    :param files: a list of dictionaries where each dictionary represents a file in the Google Drive folder.
+    :param course: a dictionary representing the parent Google Drive folder for the queried course.
+    :return: a list of strings to be sent as messages.
+    """
+    sorted_files = sorted(files, key=lambda f: f['name'])  # Sort alphabetically by filename.
+    response = [f'All of the UQAttic files for the course <{BASE_FOLDER_URL}{course["id"]} | {course["name"]}> are '
+                f'included below:\n>>> ']
+    # Handle formatting for long messages where Slack would otherwise split them.
```

Comment:
When the response is considered too long by Slack it was splitting it into several messages and those after the first were losing the block quote formatting.
Pre fix:
<img width="678" alt="screen shot 2018-06-14 at 8 47 02 pm" src="https://user-images.githubusercontent.com/9084563/41407933-60ba8604-7014-11e8-94ce-9ae2dbd42a45.png">

Post fix:
<img width="677" alt="screen shot 2018-06-14 at 8 47 17 pm" src="https://user-images.githubusercontent.com/9084563/41407954-6cd8a6aa-7014-11e8-9f9c-3db444be5b00.png">

I just thought it looked a little prettier, but maybe not at the cost of mutilating the code this much, given that it's an edge case?

Reply:
Aaaaaaah interesting. Your solution is fine then :^)

Reply:
just document this so that its clear why you're doing multiple bot posts (i.e. reference that slack would split it up, but you do it manually to retain formatting)